src/AggregateFunctions/AggregateFunctionFactory.cpp

```diff
@@ -137,7 +153,7 @@ AggregateFunctionPtr AggregateFunctionFactory::getImpl(
         if (!out_properties.returns_default_when_only_null && has_null_arguments)
             return nullptr;
 
-        return found.creator(name, argument_types, parameters);
+        return found.create(name, argument_types, parameters, query_context->getSettingsRef());
```

Comment:
See tests under UBSan: `runtime error: member call on null pointer of type 'DB::Context'`

Reply:
I changed to passing `Setting *` to creator and add check if query_context initialized
https://github.com/ClickHouse/ClickHouse/pull/22762/files#diff-4a4dba9eabf5c6708dbc69aaf1869432beb57a38415b3ffa62316e78504fff66R141-R142